### ISSUE TITLE
severity get randomly shifted when specify warn list

### ISSUE THREAD

**Issue by braindevices** [ansible/ansible-lint#2609]
##### Summary
Before 6.7 the severity does not change at all when warn list is specified. In 6.8 the severity do change with warn list. However, the severity of rules are not in the warn list gets randomly shifted.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.13.4]
ansible-lint --version
ansible-lint 6.8.2 using ansible 2.13.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Ubuntu 20.04

##### STEPS TO REPRODUCE

```console (paste below)
$ mkdir -p test-ansible-lint/tasks
$ mv rule-test.yaml test-ansible-lint/tasks/
$ ansible-lint -f json /tmp/test-ansible-lint > no-warn-list.json
$ ansible-lint -f json -w name[missing] /tmp/test-ansible-lint > warn-list.json
$ cat no-warn-list.json | jq -r '.[]|(.check_name + ":" + .severity)'
role-name:info
fqcn[action-core]:major
name[missing]:major
fqcn[action-core]:major
name[casing]:info
risky-file-permissions:info
$ cat warn-list.json | jq -r '.[]|(.check_name + ":" + .severity)'
role-name:critical
fqcn[action-core]:major
name[missing]:info
fqcn[action-core]:major
name[casing]:major
risky-file-permissions:blocker
```
content of rule-test.yaml:
```yaml
- debug:
    msg: "hello"
- name: test jinja space
  copy:
    content: |
      {% for x in somelist %}
      {{ x }}
      {% endfor %}
    dest: "/tmp/test.jinja.txt"

```

##### Desired Behavior

Only the `name[missing]` should change from major to info. All other rules' severity should remain the same.

##### Actual Behavior

all severity are randomly shifted:
```
$ diff <(cat no-warn-list.json | jq -r '.[]|(.check_name + ":" + .severity)') <(cat warn-list.json | jq -r '.[]|(.check_name + ":" + .severity)')
1c1
< role-name:info
---
> role-name:critical
3c3
< name[missing]:major
---
> name[missing]:info
5,6c5,6
< name[casing]:info
< risky-file-permissions:info
---
> name[casing]:major
> risky-file-permissions:blocker
```

**Comment by ssbarnea** [comment]
That might be a real bug, as we do not want to alter severity based on warning level. Historically we did this at some point, but we do not plan to do it anymore.

**Comment by braindevices** [comment]
if we do not modify the severity based on warning level. All the LSP will break, because the warn list settings has no effect ...

**Comment by ssbarnea** [comment]
@braindevices Last update on LSP sorts this, to be released soon.
